### ISSUE TITLE
TTS Speech Synthesiser Option for Blind Access (Votrax Emulation)

### ISSUE THREAD

**Issue by oneswitch** [mattgodbolt/jsbeeb#508]
[Emulated MAME Votrax and BBC Micro](https://youtu.be/neVZg_gy2_I?si=WGYef-gi84O3NIUt)

Pernod at StarDot shared a proof of concept where an emulated BBC Micro could output screen text (via *FX3,1) at 9600 baud to an emulated Votrax speech synthesiser. This gives a period correct emulation of one if not the first speech synthesiser people would have had access to on the BBC Micro.

His demo is of the text adventure Lords of Time speaking in the classic Gorf/Berzerk voice heard around arcades, but also commonly in use for early electronic assistive technology. He is using MAME's emulation of the Votrax SC-01 chip, with the Beebenabling serial VDU output with *FX3,1 to send all screen text to it at 9600 baud.

A bit more info here: [1983 BBC Micro with Votrax info](https://www.oneswitch.org.uk/OS-REPOSITORY/MUSEUM/VARIOUS/BLINDNESS/1983_Acorn_User_Tom_Vincent.pdf)

[MAME Votrax repository - Pernod will know more](https://github.com/mamedev/mame/tree/master/src/devices/sound)

[Some Blind Accessible Titles to test with - marked with an asterisk](https://www.oneswitch.org.uk/art.php?id=386#contentBlindAccess)

**Comment by Pernod70** [comment]
The emulation of the Votrax Type 'N Talk is at [/src/devices/machine/votraxtnt.cpp](https://github.com/mamedev/mame/blob/master/src/devices/machine/votraxtnt.cpp). 

It'll be alot of work as you'll need to emulate not only the Votrax SC-01A at [/src/devices/sound/votrax.cpp](https://github.com/mamedev/mame/blob/master/src/devices/sound/votrax.cpp) but also the M6802 at [/src/devices/cpu/m6800/m6800.cpp](https://github.com/mamedev/mame/blob/master/src/devices/cpu/m6800/m6800.cpp), obviously you'll already have the 6850 ACIA emulated.

**Comment by oneswitch** [comment]
Thanks Pernod70. Alternatively, the next best thing, and in some ways, better, would be if the screen text could be output to the in-built speech synth available. That's a far clearer voice, although not period correct.

There's definitely an audience out there for text access, plus it's a fun thing for people generally: [AudioGames.net forum with people enjoying a modified version of Pontoon from 1983](https://forum.audiogames.net/topic/56871/pontoon-game-for-the-blind-1983/)

Woud be amazing to be able to give people a link and they're straight in with all the BBC Micro games made blind accessible. 

Also realise, all this is done for free and for the love it in free time, so no expectation for anything. Cheers for all that's done so far.

**Comment by oneswitch** [comment]
No luck there changing the voice, as the settings don't exist in my version of Chrome. I tried changing in Windows, but that's already default Hazel, and English voice. No big problem. Still fantastic that it will be possible.

One thing, I spotted, is that if I select the microphone to come on, it doesn't seem to be working, and a debugger window pops up. 

[Micro Mike .SSD](https://www.dropbox.com/scl/fi/9ztlwp2cu9wd81bjy57dq/MICRO-MIKE.SSD?rlkey=kyokfhurh26mn82v1etjnonk3&dl=0)

It seems that it could be tidied up further so that it's just microphon on or off. Not the drop-down menu which I find confusing.

**Comment by oneswitch** [comment]
Chrome now working for me to change the voice to a British accent: 

1. Go to: chrome://settings/languages
2. Have the top option for the top setting as English (United Kingdom) - and ticked to "Display chrome in this language".

Worked for me.


What I am finding with the [RCEVH Scren REader](https://www.dropbox.com/scl/fi/3acycpz7yvaqvhbwbpjl2/CENTRE-TALKING-SCREEN-READER.SSD?rlkey=oe83nb0ym54nl56mq5zo5r1jy&dl=0) is that I can boot into it, use TAB to enter screen reading (e.g. using U,D,C,W,L commands), but can't enter a # hash character to enter the screen reader settings. I've tried every key on my keyboard and also an On Screen Keyboard. 

It's amazing to have a way to (partly) experience this very early fully fledged screen reader, even though it's missing that key option, and is quite slow compared to the real hardware on a BBC Micro and Votrax. At boot up it also spells out "BASIC" instead of saying BASIC, which is different behaviour for some reason.

Anyway, realise this is ultra-niche, but huge thanks for even getting it this far. Much appreciated.

Oh, and [RCEVH Screen Reader insttructions here](https://www.oneswitch.org.uk/jsbeeb/manuals/RCEVH_Screen_Reader/RCEVH_SCREEN_READER_HELP.pdf)
